prefect2_api/workspace.go

```diff
@@ -0,0 +1,75 @@
+package prefect2_api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+func (c *Client) GetAllWorkspaces() ([]Workspace, error) {
+	req, err := http.NewRequest("POST", fmt.Sprintf("%s/accounts/%s/workspaces/filter", c.PrefectApiUrl, c.PrefectAccountId), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := c.doRequest(req, c.PrefectApiKey)
+	if err != nil {
+		return nil, err
+	}
+
+	workspaces := []Workspace{}
+
+	err = json.Unmarshal(body, &workspaces)
+	if err != nil {
+		return nil, err
+	}
+
+	return workspaces, nil
+}
+
+func (c *Client) GetWorkspace(workspaceID string) (*Workspace, error) {
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/accounts/%s/workspaces/%s", c.PrefectApiUrl, c.PrefectAccountId, workspaceID), nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := c.doRequest(req, c.PrefectApiKey)
+	if err != nil {
+		return nil, err
+	}
+
+	workspace := Workspace{}
+	err = json.Unmarshal(body, &workspace)
+	if err != nil {
+		return nil, err
+	}
+
+	return &workspace, nil
+}
+
+func (c *Client) CreateWorkspace(workspace Workspace) (*Workspace, error) {
+	rb, err := json.Marshal(workspace)
```

Comment:
suggestion: You can use `json.NewEncoder()` to create a stream and pass that to `NewRequest` instead. This has better performance because it avoids memory copies.

Reply:
Done!